runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/AbstractWindmillStream.java

```diff
@@ -167,14 +185,25 @@ private StreamObserver<RequestT> requestObserver() {
   }
 
   /** Send a request to the server. */
-  protected final void send(RequestT request) {
-    lastSendTimeMs.set(Instant.now().getMillis());
-    synchronized (this) {
-      if (streamClosed.get()) {
-        throw new IllegalStateException("Send called on a client closed stream.");
+  protected final synchronized void send(RequestT request) {
```

Comment:
Generally `void method {synchronized{}}` is better than `synchronized void method{}`. Keeping synchronized as part of the method signature tells the user that this method is synchronized on `this`, and allows the user to participate in the synchronization by locking on the object's reference. Also to remove synchronized from the signature, we need to make sure no one is relying on the method being synchronized.

Here I would keep it inside, unless it needs to be at the signature level for other reasons.

Reply:
> allows the user to participate in the synchronization by locking on the object's reference

```
class A {
    synchronized void methodM() {}; 
}

A a = new A();

synchronized(a) { // here you get the same lock, that guards methodM
} 
```

Reply:
done